compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -1478,6 +1478,8 @@ pub enum BinOp {
     BitAnd,
     /// The `|` operator (bitwise or)
     BitOr,
+    /// Like `BitOr` and `BitXor`, but UB if results don't match.
```

Comment:
```suggestion
    /// Like `BitOr`, but UB if any bit is set both in the left and right operand (`left & right != 0`).
```
Let's express the requirement directly, in a way that matches the name, rather than indirectly.